synapse/logging/opentracing.py

```diff
@@ -238,18 +238,46 @@ class _DummyTagNames(object):
 # Util methods
 
 
-def only_if_tracing(func):
-    """Executes the function only if we're tracing. Otherwise return.
-    Assumes the function wrapped may return None"""
+def only_if_tracing(func=None, ret=None):
+    """Executes the function only if we're tracing. Otherwise returns `ret`."""
+
+    def only_if_tracing_inner_1(func):
+        @wraps(func)
+        def only_if_tracing_inner_2(*args, **kwargs):
+            if opentracing:
+                return func(*args, **kwargs)
+            else:
+                return ret
+
+        return only_if_tracing_inner_2
+
+    if func:
+        return only_if_tracing_inner_1(func)
+    else:
+        return only_if_tracing_inner_1
+
+
+def ensure_active_span(message, ret=None):
+    """Executes the operation only if there is an active span. Otherwise it logs
+    an error and returns 'ret'."""
+
+    def ensure_active_span_inner_1(func):
+        @wraps(func)
+        def ensure_active_span_inner_2(*args, **kwargs):
+            if not opentracing.tracer.active_span:
+                logger.error(
+                    "There was no active span when trying to %s."
+                    " Did you forget to start one or did a context slip?",
+                    message,
```

Comment:
```suggestion
                    message,
                    stack_info=True,
```

So it also adds a strack trace, which would probably be useful, maybe